app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/StartController.scala

```diff
@@ -69,80 +69,95 @@ class StartController @Inject()(
     with SessionUpdates
     with IvBehaviour {
 
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
   def start(): Action[AnyContent] = authenticatedActionWithRetrievedDataAndSessionData.async { implicit request =>
-    (
-      request.authenticatedRequest.userType,
-      request.sessionData.flatMap(_.journeyStatus)
-    ) match {
+    updateSession(sessionStore, request)(_.copy(userType = request.authenticatedRequest.userType)).flatMap {
```

Comment:
do we still need this `updateSession` here if you are populating the user type in the session data in the handle methods now?

Reply:
yes, for the cases where we don't go into the handlers.

Reply:
the only time we don't go into a handle method is if there is session data already there - in those instances we shouldn't need to rewrite the user type

Reply:
both you comments are connected - if we don't the update in the SessionDataACtionWithRetrievedData then we need to do this update in the start controller. If we don't do the update StartController then we need to do the update in the SessionDataActionWithRetrievedData. I will take it out of the StartController.

Reply:
cool - making the `SessionData` non-optional in `RequestWithSessionDataAndRetrievedData` should make some of the code later on easier as well then